### PR TITLE
[Tooling] Migrate all lanes to keyword-params syntax

### PR DESCRIPTION
**Pull request by AliSoftware**
📘 Related to Project Thread: paaHJt-78B-p2

> [!IMPORTANT]
> This PR stacks on top of https://github.com/Automattic/pocket-casts-ios/pull/2216 (which itself sits on top of #2215) and should only be merged _after_ those have been merged first, to not pollute their diffs.
>
> ```
> release-on-ci/keywords-params
>   |
>   |-- [#2217] (this PR)
>   v
> release-on-ci/reorder-lanes
>   |
>   |-- [#2216]
>   v
> release-on-ci/refactor-code-freeze
>   |
>   |-- [#2215]
>   v
> trunk
> ```

## What

This PR migrates all the lanes that were using `do |options| …` to use keyword params syntax instead (e.g. `do |skip_confirm: false| …`)

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -361,10 +359,10 @@ platform :ios do
   #  - Removes branch protection and close milestone
   #  - Triggers the final release on CI
   #
-  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  # @param skip_confirm [Boolean] If true, avoids any interactive prompt (default: false)
```
Comment (dangermattic) [inline fastlane/Fastfile:362]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/pocket-casts-ios/blob/cc9ad91bdf6900c30acab5384a172d1680c3d9a8/fastlane/Fastfile#L362"></span>Layout/TrailingWhitespace: Trailing whitespace detected.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 2]
```diff
@@ -483,42 +480,42 @@ platform :ios do
...
 
   # Builds the Pocket Casts app and uploads it to TestFlight, for beta-testing or final release
   #
-  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  # @option [Boolean] skip_prechecks (default: false) If true, don't run the prechecks and ios_build_preflight
-  # @option [Boolean] create_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
-  # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
+  # @param beta_release [Boolean] If true, the GitHub release will be marked as being a pre-release
+  # @param skip_prechecks [Boolean] If true, don't run the prechecks and ios_build_preflight (default: false)
+  # @param skip_confirm [Boolean] If true, avoids any interactive prompt (default: false)
```
Comment (dangermattic) [inline fastlane/Fastfile:501]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/pocket-casts-ios/blob/cc9ad91bdf6900c30acab5384a172d1680c3d9a8/fastlane/Fastfile#L501"></span>Layout/TrailingWhitespace: Trailing whitespace detected.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 3]
```diff
@@ -850,14 +846,13 @@ inal copy, and try again.")
 
   # Upload the localized metadata (from `fastlane/metadata/`) to App Store Connect
   #
-  # @option [Boolean] with_screenshots (default: false) If true, will also upload the latest screenshot files to ASC
+  # @param with_screenshots [Boolean] If true, will also upload the latest screenshot files to ASC (default: false)
```
Comment (dangermattic) [inline fastlane/Fastfile:849]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/pocket-casts-ios/blob/cc9ad91bdf6900c30acab5384a172d1680c3d9a8/fastlane/Fastfile#L849"></span>Layout/TrailingWhitespace: Trailing whitespace detected.</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/60">7.74</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by spencertransier — APPROVED** [review]

**Review by SergioEstevao — APPROVED** [review]
🚢

File: fastlane/Fastfile [hunk 4]
```diff
@@ -178,7 +178,7 @@ platform :ios do
   end
 
   desc 'Download and configure code signing certificates and provisioning profiles for the App Store build.'
-  lane :configure_code_signing_app_store do |options|
+  lane :configure_code_signing_app_store do |readonly: true|
```
Comment (iangmaia) [inline fastlane/Fastfile:181]: The first thing that comes to mind is that in other repos, we also replaced `desc` with YARD docs.
This PR already looks good to me as the main change are the keyword parameters, but I wonder if we should also standardize the comments.

**Review by iangmaia — APPROVED** [review]
Small comment that, if needed, doesn't have to be addressed in this PR.

File: fastlane/Fastfile (hunk 4 above)
Comment (AliSoftware) [reply fastlane/Fastfile:181]: That's a good point. I'll take note to clean that up in a future PR (I already planned to make another pass at the YARD doc of all the lanes at some point anyway) 👍

**Pull request merged**